.github/workflows/ci-main.yml

```diff
@@ -53,37 +76,44 @@ jobs:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Build
+        if: matrix.rust == 'stable'
         uses: actions-rs/cargo@v1
         with:
           command: build
           args: --all-features
 
       - name: Test
+        if: matrix.rust == 'nightly'
         uses: actions-rs/cargo@v1
-        env: 
+        env:
           SOURCE_STELLAR_SECRET_MAINNET: ${{ secrets.SOURCE_STELLAR_SECRET_MAINNET }}
           SOURCE_STELLAR_SECRET_TESTNET: ${{ secrets.SOURCE_STELLAR_SECRET_TESTNET }}
           DEST_STELLAR_SECRET_MAINNET: ${{ secrets.DEST_STELLAR_SECRET_MAINNET }}
           DEST_STELLAR_SECRET_TESTNET: ${{ secrets.DEST_STELLAR_SECRET_TESTNET }}
         with:
+          toolchain: nightly-2024-02-09
           command: test
           args: --all --all-features
 
       - name: Rustfmt
+        if: matrix.rust == 'nightly'
         uses: actions-rs/cargo@v1
         with:
+          toolchain: nightly-2024-02-09
           command: fmt
           args: --all -- --check
 
-      - name: Clippy -- Main
+      - name: Clippy -- Libraries and Binaries
+        if: matrix.rust == 'stable'
         uses: actions-rs/cargo@v1
         with:
           command: clippy
-          args: --all-features -- -W clippy::all -A clippy::style -A clippy::forget_copy -A clippy::forget_ref
+          args: --all-features --lib --bins -- -W clippy::all -A clippy::style -A forgetting_copy_types -A forgetting_references
 
-      - name: Clippy -- All Targets
+      - name: Clippy -- Tests and Examples
+        if: matrix.rust == 'nightly'
         uses: actions-rs/cargo@v1
         with:
+          toolchain: nightly-2024-02-09
           command: clippy
-          # We are a bit more forgiving when it comes to the code in tests and only check for correctness
-          args: --all-features --all-targets -- -A clippy::all -W clippy::correctness -A clippy::forget_copy -A clippy::forget_ref
\ No newline at end of file
+          args: --all-features --tests --benches --examples -- -A clippy::all -W clippy::correctness -A forgetting_copy_types -A forgetting_references
```

Comment:
we cannot use `--all-targets` here because the `test` must use the nightly version, while the libs and bins must use the stable version